dali/kernels/erase/erase_gpu.h

```diff
@@ -0,0 +1,530 @@
+  ivec<ndim> sample_shape;
+  ivec<ndim> sample_stride;
+};
+
+template <int ndim>
+DALI_HOST_DEV ivec<ndim> get_region_start(int region_idx, ivec<ndim> region_shape,
+                                          ivec<ndim> sample_shape) {
+  // let's calculate how many regions cover given axes:
+  auto region_cover_shape = div_ceil(sample_shape, region_shape);
+  // this introduces the same ordering of regions as data layout
+  auto region_cover_strides = GetStrides(region_cover_shape);
+  ivec<ndim> region_position{};  // n-dim index of region among other regions
+  region_position[0] = region_idx / region_cover_strides[0];
+  for (int i = 1; i < ndim; i++) {
+    region_idx -= region_position[i - 1] * region_cover_strides[i - 1];
+    region_position[i] = region_idx / region_cover_strides[i];
+  }
+  return region_position * region_shape;
+}
+
+template <int ndim>
+__device__ ibox<ndim> get_region(int region_idx, ivec<ndim> region_shape, ivec<ndim> sample_shape) {
+  auto anchor = get_region_start(region_idx, region_shape, sample_shape);
+  return {anchor, anchor + region_shape};
+}
+
+constexpr unsigned FULL_MASK = 0xffffffffu;
+
+/**
+ * @brief Returns true if that parameter configuration is just outer layer of recursive loop call
+ */
+constexpr bool select_outer_loops(int ndim, int current_dim, int channel_dim) {
+  // channel_dim is not one of the 2 last dimensions
+  if (channel_dim < ndim - 2) {
+    return ndim - current_dim > 2;  // true if we are not in the last two dimensions
+  } else {
+    return ndim - current_dim > 3;  // we have to select a variant that accommodates channel dim
+  }
+}
+
+
+/**
+ * @brief Outer layer of recursive loop over current region
+ */
+template <typename Worker, int channel_dim, int current_dim = 0, typename T, int ndim>
+__device__ std::enable_if_t<select_outer_loops(ndim, current_dim, channel_dim)>
+erase_generic(erase_sample_desc<T, ndim> sample, ibox<ndim> region, span<T> fill_values,
+              span<ibox<ndim>> erase_regions = {}, ivec<ndim> coordinate = {},
+              int64_t offset_base = 0) {
+  constexpr int d = current_dim;
+  int boundary = ::min(region.hi[d], sample.sample_shape[d]);
+  for (int i = region.lo[d]; i < boundary; i++) {
+    coordinate[d] = i;
+    int64_t offset = offset_base + i * sample.sample_stride[d];
+    erase_generic<Worker, channel_dim, d + 1>(sample, region, fill_values, erase_regions,
+                                              coordinate, offset);
+  }
+}
+
+/**
+ * We have 2 full most inner loops, channel_dim is somewhere on the way here, or doesn't exist
+ *
+ * We handle this by covering the innermost dimensions by:
+ * for (y : blockDim.y)
+ *   for (x : blockDim.x)
+ *     process {coordinate[0], coordinate[1], ..., coordinate[current_dim - 1], y, x}
+ */
+template <typename Worker, int channel_dim, int current_dim = 0, typename T, int ndim>
+__device__ std::enable_if_t<ndim - current_dim == 2 && channel_dim < ndim - 2>
```

Comment:
```suggestion
__device__ std::enable_if_t<ndim - current_dim == 2 && (channel_dim < ndim - 2)>
```

Reply:
Why this and not 
```suggestion
__device__ std::enable_if_t<(ndim - current_dim == 2) && (channel_dim < ndim - 2)>
```
or
```suggestion
__device__ std::enable_if_t<(ndim - current_dim == 2 && channel_dim < ndim - 2)>
```